Mailgun: add configuration for EU server (#41267)
home-assistant/home-assistant.github.io: source/_integrations/mailgun.markdown

  description: This is the API token that has been generated in your Mailgun account.
  required: true
  type: string
eu_domain:

>   description: This configuration allows for the usage of Mailgun EU server. If set to true, it will use EU server instead of US.

Reviewer: ```suggestion
  description: This configuration allows you to use the Mailgun EU server. If set to true, it will use the EU server instead of the US server.
```
